src/validate.js

```diff
@@ -18,18 +18,22 @@ export default function validate(descriptor
   const remoteString = remoteProfiles.toString()
 
   if (PROFILES_CACHED[remoteString]) {
-    return new Promise(resolve => {
+    return new Promise((resolve, reject) => {
       PROFILES_CACHED[remoteString].then(profiles => {
         resolve(profiles.validate(descriptor, profile))
+      }).catch(err => {
+        reject(err)
```

Comment:
We need this rejections to happen in the scenario when `remote` profiles are being used but connection can't be made.